dashboard/src/components/ParkingRegionsMap.tsx

```diff
@@ -0,0 +1,204 @@
+                />
+                <ReactLeaflet.ScaleControl
+                    metric={true}
+                    imperial={false}
+                    maxWidth={150}
+                />
+                {geoJsonElement}
+            </Map>);
+    }
+
+    private bindPopupToRegion = (
+        region: Region,
+        layer: Leaflet.Layer
+    ): void => {
+        const p = region.properties;
+        if (!p) {
+            return;
+        }
+        const capacityEstimateFromArea = Math.round(p.areaKm2 * 1000);
+        const capacity = p.capacityEstimate || capacityEstimateFromArea;
+        const count = p.parkingCount || 0;
+        const parkingsPerKm2 = count / p.areaKm2;
+        const usagePercentage = 100 * count / capacity;
+        layer.on('click', (event) => {
+            if (this.props.onRegionClicked) {
+                this.props.onRegionClicked(region);
+            }
+        });
+        layer.bindTooltip(p.name);
+        const text = `
+            <h3>${p.name}</h3>
+            <b>Pinta-ala</b>:
+              ${p.areaKm2.toFixed(2)} km<sup>2</sup><br>
+            <b>Arvioitu kapasiteetti</b>:
+              ${p.capacityEstimate}<br>
+            <b>Arvioitu kapasiteetti (pinta-ala)</b>:
+              ${capacityEstimateFromArea}<br>
+            <b>Paikkoja per km<sup>2</sup></b>:
+              ${p.spotsPerKm2.toFixed(2)}<br>
+            <hr>
+            <b>Pysäköintejä</b>:
+              ${p.parkingCount || 0}<br>
+            <b>Pysäköintejä per km<sup>2</sup></b>:
+              ${parkingsPerKm2.toFixed(2)}<br>
+            <b>Käyttöaste</b>:
+              ${usagePercentage.toFixed(2)}&nbsp;%<br>
+            `;  /* + '<pre>' + JSON.stringify(p, null, 2) + '</pre>'*/
+        layer.bindPopup(text);
+    }
+
+    private handleViewportChange = (viewport: Viewport) => {
+        if (!this.map || !this.props.onViewportChanged) {
+            return;
+        }
+
+        const boundsObj = this.map.leafletElement.getBounds();
+        const ne = boundsObj.getNorthEast();
+        const sw = boundsObj.getSouthWest();
+        const bounds = {
+            neLat: ne.lat, neLng: ne.lng,
+            swLat: sw.lat, swLng: sw.lng};
+        const {center, zoom} = viewport;
+        this.props.onViewportChanged({bounds, center, zoom});
+    }
+}
+
+function getKeyForRegions(regions: Region[]): string {
+    if (!regions) {
+        return '';
+    }
+    if (!regions.length) {
+        return 'empty';
+    }
+    let lengthSum = 0;
+    let selectedId = '';
+    for (let region of regions) {
+        if (region.properties) {
+            lengthSum += Object.keys(region.properties).length || 0;
+            lengthSum += region.properties.parkingCount || 0;
+            if (region.properties.isSelected) {
+                selectedId = region.id;
+            }
+        }
+    }
+    const lastRegion = regions[regions.length - 1];
+    return (
+        regions.length.toString() + '/'
```

Comment:
Consider using ` for string/variable concatenation, in this case:
```
`${regions.length.toString()}/${lengthSum}/${lastRegion.id}/${selectedId }`
```

Reply:
Good suggestion. Thanks!